Seqr metadata templates and instructions

EddieLF opened the pull request:
So these are the updated templates and instruction documents. Happy to hear any feedback on everything as we want this to be clear and understandable to any collaborator who is linked to this repo to download the templates. I'll add readmes for the subdirectories in this repo too. Also have a look through the instructional documents and see if they are 100% clear and note any changes you think should be made.

For the templates themselves, only a few changes have been made:
- 'Individuals_template' renamed to 'pedigree_metadata_template' for clarity.
- 'individuals_metadata', 'families_template', 'sample_mapping' have been renamed to 'individuals_metadata_template', 'families_metadata_template', 'sample_mapping_template' respectively.
- 'Notes' field removed from pedigree metadata in favour of the 'individual notes' in the individuals metadata 
- Additional fields and allowed values included in individuals_metadata_template / data dictionary
- Excel, csv, and tsv versions available

Once we're happy with this and merge it into the main branch, I'll go back and update the links to the templates in the PDF to the actual locations in this repo.

EddieLF committed:
Seqr metadata templates and instructions

EddieLF committed:
linter fix for some reason

cassimons commented:
Thanks, Eddie. I think the changes to the templates and template names are well overdue. The help docs are great, but I would be keen to have them not as word docs/pdfs, but as standard github markdown pages. That way people are always interacting with the 'live' version, not a copy that was sent to them 6 months ago.  Can we try reformating them>

illusional reviewed (commented):
Looking good, +1 to Cas' comment. Could we also avoid directories with spaces in the name.
And you could simplify the pedigree to `pedigree_template.ext` (removing _metadata_ from the name).

SamBryen commented:
Looks great @EddieLF! 
It might be useful to include guidelines about what to do when you're adding family members to an existing family or adding new/additional sequencing data for individuals that have already been included in previously submitted pedigrees. There have been a couple of instances where an individual has been added to metamist under two different individual IDs because they were included in the initial pedigree for a family before sequencing data was available. Later when data is available, it is sent to us with a new pedigree file that contains typos or completely new IDs because the collaborators are unaware that they need to perfectly match pre-existing IDs sent. A suggestion:

"Note: The Individual_ID is used by CPG to internally track individuals. If a new Individual_ID is provided in the metadata, a new individual will be created within our system. If providing new data for individuals that have already been included in metadata previously sent to CPG, please use the exact same Individual_ID." 
Maybe include details about if new information provided would override older information or if they can keep fields blanks if already submitted so as to not overwrite previous entries?

I also noticed that on page 3 it states that we prefer to receive CRAMs over BAMs or FASTQs. @illusional I had in my head that we preferred FASTQs because that's what the pipeline required and we needed to convert any CRAMs or BAMs back to FASTQs before we could do anything with them?

cassimons commented:
> we preferred FASTQs because that's what the pipeline required and we needed to convert any CRAMs or BAMs back to FASTQs before we could do anything with them?

In addition to the reduced compute, I prefer fastqs as there is less chance that reads have been lost in the original mapping platform (seen it many times).

EddieLF committed:
added readme and images:

EddieLF committed:
added pedigree warning and formatting changes

EddieLF committed:
update directory names

EddieLF committed:
removing old directories

EddieLF committed:
move readme to templates folder

EddieLF committed:
moved images folder

EddieLF committed:
added data transfer readme

EddieLF committed:
lint fix

EddieLF committed:
lint fixes again

EddieLF committed:
added base readme, relative links, and excel data validation rules

EddieLF committed:
removed pdfs and reformatted readme and data dict

EddieLF committed:
markdownlint changes

EddieLF commented:
I've now updated this branch with:

1. A base README, containing links to the 'Uploading data' README, the 'Organising metadata' README, and a link to the metadata templates folder.
2. All READMEs contain relative links to the various templates and other READMEs 
3. Excel templates have data validation rules so collaborators must select from a drop down list to fill out certain fields (e.g. only 0, 1, or 2 allowed in the Sex field of the pedigree template, only TRUE/FALSE or blank allowed in the Consanguinity field of the individual metadata template)
5. Incorporated the feedback from your wonderful comments:
 - removed PDFs and docx files, 
 - removed spaces from directory names, 
 - removed "metadata" from pedigree_template, 
 - added note about repeat individuals, 
 - corrected note about fastqs > CRAMs/BAMs file preference

Let me know if there's anything else you think should be changed, thanks!

cassimons commented:
@SamBryen would you mind trying out the excel templates "as a user" and check if the validation works for you?

cassimons commented on resources/seqr_metadata_templates/README.md:
```diff
@@ -0,0 +1,275 @@
+
+### i. Purpose
+
+The purpose of this document is to provide instructions on how to prepare project specific data and metadata for use in seqr<br /> related to an individual's family history (pedigree).
```
```suggestion
To enable the CPG to process and display your genomic data in seqr we require some additional information about the genomic data and the individuals it was generated from. This document provides instructions on how to prepare project-specific data and metadata for use in seqr.
```

cassimons commented on resources/seqr_metadata_templates/README.md:
```diff
@@ -0,0 +1,275 @@
+
+### ii. Background
+
+The CPG utilises four distinct metadata files to provide information about samples to the variant curation team as they <br /> perform variant analysis in seqr.
```
Do we need the inline `<br />`s? It will handle wrapping natively wont it?

EddieLF replied:
Yeah, I think I got a bit fixated on making everything appear nice in the markdown preview while I was writing this up that I completely forgot about all the different devices and screen sizes this will be viewed from. 
I'll get rid of some of these since they will look silly with the native wrapping in there as well

cassimons commented on resources/seqr_metadata_templates/README.md:
```diff
@@ -0,0 +1,275 @@
+<br />
+
+#### **Table 1:** Definitions of the metadata files used in rare disease projects <!-- omit from toc -->
+
```
Can we add a column with direct download links to each file? eg something like:
```
| Download |
| [csv](url) <br /> [excel](url) |
```

cassimons commented on resources/seqr_metadata_templates/README.md:
```diff
@@ -0,0 +1,275 @@
+
+*A pedigree is a structured description of the phenotypical and familial relationships between samples.
+
+The CPG uses the tool ‘GATK HaplotypeCaller’, which can incorporate pedigree information in the genomic analysis of samples.
```
HaplotypeCaller does not make use of pedigree info.
```suggestion

```

cassimons commented on resources/seqr_metadata_templates/README.md:
```diff
@@ -0,0 +1,275 @@
+
+2. [csv templates](csv_templates/)<br />
+
+3. [tsv templates](tsv_templates/)<br />
```
Do we need tsv and csv or should we just pick one (presumably csv for multi-line escaping in notes fields)?

EddieLF replied:
I agree it's probably not necessary to have both. I'm happy to just reduce it to csv

cassimons reviewed (approved):
Looks good thanks @EddieLF. A few minor suggestions below.

SamBryen commented:
Looks good to me! I signed out and was able to download them all correctly. I like the links that directly navigate to the download button, I was a little worried that it wouldn't be obvious to those who have never used GitHub, but I think it's clear enough.

EddieLF committed:
updated templates readme text and wrapping, removed tsv templates

EddieLF committed:
added link to metadata templates to uploading data readme

EddieLF committed:
fix link to metadata templates in readme

EddieLF committed:
added a space somewhere

EddieLF commented on resources/seqr_metadata_templates/README.md:
```diff
@@ -0,0 +1,264 @@
+| Birth Year | Numeric year of birth. E.g. 2010 | If you have collected a DOB, e.g. 01-01-2001, please only include the **year** component. |
+| Death Year | Numeric year of death, if applicable. <br /> Leave blank otherwise. | If you have collected a DOD, e.g. 01-01-2001, please only include the **year** component. |
+| Age of Onset | **One of the following:** <br /><br />Congenital onset, <br />Embryonal onset, <br />Fetal onset, <br />Neonatal onset, <br />Infantile onset, <br />Childhood onset, <br />Juvenile onset, <br />Young adult onset, <br />Adult onset, <br />Middle age onset, <br />Late onset | *This is a rough suggestion, with no clinical source.*<br /> <br /> **Congenital onset:** conception to bir ...
+| Individual Notes | Alphanumeric characters |   |
```
Should any of these empty description fields be filled in, or are they ok blank?

cassimons replied:
> Should any of these empty description fields be filled in, or are they ok blank?

All fields are optional other than `Family ID` and `Individual ID`.

However, now that you mention it it would be good to move `Individual Notes` up to the third row and add a comment that this is a free text field that will be displayed in seqr. It can be a good place to describe the individual's presentation and any features important for variant curation.
